config/main.py

```diff
@@ -1227,6 +1227,95 @@ def validate_patch(patch):
     except Exception as e:
         raise GenericConfigUpdaterError(f"Validate json patch: {patch} failed due to:{e}")
 
+
+def multiasic_validate_single_file(filename):
+    ns_list = [DEFAULT_NAMESPACE, *multi_asic.get_namespace_list()]
+    file_input = read_json_file(filename)
+    file_ns_list = [DEFAULT_NAMESPACE if key == HOST_NAMESPACE else key for key in file_input]
+    if set(ns_list) != set(file_ns_list):
+        click.echo(
+            "Input file {} must contain all asics config. ns_list: {} file ns_list: {}".format(
+                filename, ns_list, file_ns_list)
+        )
+        raise click.Abort()
+
+
+def load_sysinfo_if_missing(asic_config):
+    device_metadata = asic_config.get('DEVICE_METADATA', {})
+    platform = device_metadata.get("localhost", {}).get("platform")
+    mac = device_metadata.get("localhost", {}).get("mac")
+    if not platform or not mac:
+        if not platform:
+            log.log_warning("platfrom is missing from Input file")
+        if not mac:
+            log.log_warning("mac is missing from Input file")
+        return True
+    return False
+
+
+def flush_configdb(namespace=DEFAULT_NAMESPACE):
+    if namespace is DEFAULT_NAMESPACE:
+        config_db = ConfigDBConnector()
```

Comment:
>ConfigDBConnector [](http://example.com/codeflow?start=20&length=17)

Is it same as the statement in `else`? if yes, no need to check namespace.

Reply:
No. There is one more difference with argument `use_unix_socket_path`. The default value is False. This part is copied from original code.